### ISSUE TITLE
https mutual authentication for imposters

### ISSUE THREAD

**Issue by etxnija** [bbyars/mountebank#66]
A request to add support for ssl mutual auth for imposters. 

I have to virtualize https services that uses mutual auth, so in order to get the correct behavior for the clients the server (mountebank imposter) need to request a client cert. 

We manage to get mb to do mutual auth with these changes. Adding a new protocol. 
```
//mountebank.js
// note added https_mutual protocol
function create (options) {
    var app = express(),
        imposters = {},
        protocols = {
            tcp: require('./models/tcp/tcpServer').initialize(options.allowInjection, !options.nomock),
            http: require('./models/http/httpServer').initialize(options.allowInjection, !options.nomock),
            https: require('./models/https/httpsServer').initialize(options.allowInjection, !options.nomock,
                                                                      options.keyfile, options.certfile),
            https_mutual: require('./models/https/httpsServer').initialize(options.allowInjection, !options.nomock,
                                                                      options.keyfile, options.certfile, true),
            soap: require('./models/soap/soapServer').initialize(options.allowInjection, !options.nomock),
            smtp: require('./models/smtp/smtpServer').initialize(!options.nomock),
            foo: require('./models/foo/fooServer').initialize(options.allowInjection, !options.nomock)
        },
...........
...........
```

And adding the `requestCert` option
```
// models/https/httpsServer.js
// added a requestClientCert & set the option requestCert
function initialize (allowInjection, recordRequests, keyfile, certfile, requestClientCert) {
    var cert = {
            key: fs.readFileSync(keyfile || __dirname + '/cert/mb-key.pem'),
            cert: fs.readFileSync(certfile || __dirname + '/cert/mb-cert.pem'),
            requestCert: !!requestClientCert
        },
        createServer = function () {
            return https.createServer(cert);
        };
    return baseHttpServer.setup('https', createServer).initialize(allowInjection, recordRequests);
}
```
 
With these changes mb imposter will request a client certificate. But it will not fail if cert is not valid don't. For that you have to set `rejectUnauthorized` to true.

**Comment by bbyars** [comment]
Great idea - I'll get it added as soon as I free up some time (likely this weekend).  Thanks!

**Comment by bbyars** [comment]
Released as part of v1.2.122
